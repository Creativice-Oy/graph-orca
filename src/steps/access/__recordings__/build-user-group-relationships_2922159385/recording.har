{
  "log": {
    "_recordingName": "build-user-group-relationships",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "60949b75b7edca4147beec378bd95467",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "app.eu.orcasecurity.io"
            }
          ],
          "headersSize": 652,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://app.eu.orcasecurity.io/api/rbac/group"
        },
        "response": {
          "bodySize": 153,
          "content": {
            "mimeType": "application/json",
            "size": 153,
            "text": "{\"version\":\"0.1.0\",\"status\":\"success\",\"data\":{\"groups\":[{\"id\":\"b3835712-ff38-4214-91f6-5a740a157e08\",\"name\":\"Test\",\"description\":\"\",\"sso_group\":false}]}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 03 Jun 2022 14:32:50 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "153"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "-"
            },
            {
              "name": "vary",
              "value": "Accept, Cookie, Origin"
            },
            {
              "name": "allow",
              "value": "GET, POST, HEAD, OPTIONS"
            },
            {
              "name": "x-orca-request-id",
              "value": "31512b8d-b82d-4f55-857b-ea944d28bf0c"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-629a1b92-51115caf72dc43ca05005fd8"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "referrer-policy",
              "value": "same-origin"
            }
          ],
          "headersSize": 387,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-06-03T14:32:50.878Z",
        "time": 102,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 102
        }
      },
      {
        "_id": "6de5f6362cedc8ab78fe74a1bc37e0d5",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "app.eu.orcasecurity.io"
            }
          ],
          "headersSize": 689,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://app.eu.orcasecurity.io/api/rbac/group/b3835712-ff38-4214-91f6-5a740a157e08"
        },
        "response": {
          "bodySize": 197,
          "content": {
            "mimeType": "application/json",
            "size": 197,
            "text": "{\"version\":\"0.1.0\",\"status\":\"success\",\"data\":{\"group\":\"Test\",\"description\":\"\",\"all_users\":false,\"users\":[{\"id\":\"5427b0b3-4d99-47cb-bf63-fee5adff787c\",\"email\":\"teemuhirvonen@contractor.jupiterone.com\",\"first_name\":\"Teemu\",\"last_name\":\"Hirvonen\"}]}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 03 Jun 2022 14:32:51 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "197"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "-"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Cookie, Origin"
            },
            {
              "name": "allow",
              "value": "GET, PUT, DELETE, HEAD, OPTIONS"
            },
            {
              "name": "x-orca-request-id",
              "value": "08d010d0-fda9-48c2-b06a-82cc59fedab5"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-629a1b93-0316b0a10b013b711a38b5f1"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "referrer-policy",
              "value": "same-origin"
            }
          ],
          "headersSize": 435,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-06-03T14:32:50.983Z",
        "time": 103,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 103
        }
      },
      {
        "_id": "5e065954e5093fa42fcfe9ea42416532",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "app.eu.orcasecurity.io"
            }
          ],
          "headersSize": 660,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://app.eu.orcasecurity.io/api/organization/users"
        },
        "response": {
          "bodySize": 1248,
          "content": {
            "mimeType": "application/json",
            "size": 1248,
            "text": "{\"status\":\"success\",\"data\":{\"name\":\"Partners\",\"users\":[{\"user_id\":\"17f081c9-5665-4668-b10e-3c91b3409761\",\"email\":\"avidor@axonius.com\",\"first\":\"Avidor\",\"last\":\"Bartov\"},{\"user_id\":\"5427b0b3-4d99-47cb-bf63-fee5adff787c\",\"email\":\"teemu.hirvonen@contractor.jupiterone.com\",\"first\":\"Teemu\",\"last\":\"Hirvonen\"},{\"user_id\":\"b10d3974-4ddc-45a4-9d46-a7475d4d50b0\",\"email\":\"tomery@bionic.ai\",\"first\":\"Tomer\",\"last\":\"Yanovich\"},{\"user_id\":\"c113b4b4-4f76-4882-a449-5d2728044f00\",\"email\":\"roip@bionic.ai\",\"first\":\"Roi\",\"last\":\"Paz\"},{\"user_id\":\"b26deb2d-9d1e-4afc-990f-ef2f639ba056\",\"email\":\"rois@google.com\",\"first\":\"roi\",\"last\":\"shemesh\"},{\"user_id\":\"55462cb3-4a68-4f62-9edc-1a35f2b24daf\",\"email\":\"bar@vulcan.io\",\"first\":\"Bar\",\"last\":\"Buskila\"},{\"user_id\":\"1ad144c7-35b1-40e4-bdce-7844d382b461\",\"email\":\"idozaifman@matttter.com\",\"first\":\"ido\",\"last\":\"zaifman\"},{\"user_id\":\"c6c46b65-76b0-4488-8ee2-f9c4c47c3def\",\"email\":\"eyal@bionic.ai\",\"first\":\"Eyal\",\"last\":\"Mamo\"},{\"user_id\":\"28bc0909-820b-4250-825e-ccea3e6c8e98\",\"email\":\"carolineg@bionic.ai\",\"first\":\"Caroline\",\"last\":\"Gal\"},{\"user_id\":\"c3838b95-d326-4d47-a219-0328f845a1ac\",\"email\":\"liorm@bionic.ai\",\"first\":\"Lior\",\"last\":\"Mazza\"},{\"user_id\":\"d59a7d31-ece1-4dc8-a039-6658cb40816e\",\"email\":\"guy@anecdotes.ai\",\"first\":\"Guy\",\"last\":\"Boyarski\"},{\"user_id\":\"f94bcca1-7807-4684-8b94-9019c50de53f\",\"email\":\"ilan@vulcan.io\",\"first\":\"Ilan\",\"last\":\"Spector\"},{\"user_id\":\"da080549-396f-4cea-829c-f46c026bc9e4\",\"email\":\"shahar.kgn@gmail.com\",\"first\":\"shahar\",\"last\":\"kagan\"},{\"user_id\":\"508bdd4f-c791-4f55-9ca7-f3f9470104e7\",\"email\":\"tip.labops@siemplify.co\",\"first\":\"Milen\",\"last\":\"Ivanov\"},{\"user_id\":\"c37270f5-d03f-482c-bd37-ae4aa543c2c7\",\"email\":\"austin.kelleher@jupiterone.com\",\"first\":\"Austin\",\"last\":\"Kelleher\"},{\"user_id\":\"c5876af0-4145-487b-84e7-a15a0e823ab6\",\"email\":\"sam.andrus@jupiterone.com\",\"first\":\"Sam\",\"last\":\"Andrus\"},{\"user_id\":\"8f19d961-31b1-4313-869f-3792280c5d14\",\"email\":\"andrewkundik@gmail.com\",\"first\":\"Andrei\",\"last\":\"Kundzik\"},{\"user_id\":\"0c397cfa-8fa4-4101-9e77-a99ad75f3508\",\"email\":\"christeen@vulcan.io\",\"first\":\"Christeen\",\"last\":\"Khshiboun\"},{\"user_id\":\"8800d202-546a-47bc-9e7b-72f1a7163c88\",\"email\":\"nino@vulcan.io\",\"first\":\"Nino\",\"last\":\"klimiashvili\"},{\"user_id\":\"b816c873-fd88-4507-90cf-9144edc677e9\",\"email\":\"erica.nagle@jupiterone.com\",\"first\":\"Erica\",\"last\":\"Nagle\"},{\"user_id\":\"73a13344-7fab-4906-abe3-d19255ee3aed\",\"email\":\"wayne_johnstone@rapid7.com\",\"first\":\"Wayne\",\"last\":\"Johnstone\"},{\"user_id\":\"d912b23d-b1cf-43aa-9a90-f70a757210f9\",\"email\":\"bindu_laxminarayan@rapid7.com\",\"first\":\"Bindu\",\"last\":\"Laxminarayan\"}],\"pending_invites\":[{\"email\":\"integrationalliance@rapid7.com\",\"role\":\"Editor\",\"all_cloud_account\":true,\"cloud_accounts\":[]},{\"email\":\"ilya@bionic.ai\",\"role\":\"Administrator\",\"all_cloud_account\":true,\"cloud_accounts\":[]},{\"email\":\"stefan.virag@contractor.jupiterone.com\",\"role\":\"Administrator\",\"all_cloud_account\":true,\"cloud_accounts\":[]}]}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 03 Jun 2022 14:32:51 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "1248"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "-"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Cookie, Origin"
            },
            {
              "name": "allow",
              "value": "GET, POST, DELETE, HEAD, OPTIONS"
            },
            {
              "name": "x-orca-request-id",
              "value": "58cf29c3-d5db-4c6f-8e9b-3ce94e589924"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-629a1b93-1ca36ce853b5e8ac400e3424"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "referrer-policy",
              "value": "same-origin"
            }
          ],
          "headersSize": 437,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-06-03T14:32:51.089Z",
        "time": 143,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 143
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
