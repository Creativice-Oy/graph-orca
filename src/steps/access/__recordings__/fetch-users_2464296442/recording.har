{
  "log": {
    "_recordingName": "fetch-users",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "6075ce7e2350e7b54c2efca26cba0f39",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 174,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "174"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.orcasecurity.io"
            }
          ],
          "headersSize": 273,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "{\"security_token\":\"gE0s81evhtDZdEragMNC7qUYAfjD0CM13QBY7NxWctZXZGLZyAzRysL4xRmtZ9bVNeEqmBZWDkwFlyb0Y4BVr4lc21gWzvpcFFzRXqWFwa9ItaAGvMOYdugcuRoftp07V7g6GxMYS4xAiJ6Fg1JOud1k4\"}"
          },
          "queryString": [],
          "url": "https://api.orcasecurity.io/api/user/session"
        },
        "response": {
          "bodySize": 722,
          "content": {
            "mimeType": "application/json",
            "size": 722,
            "text": "{\"status\":\"success\",\"role\":\"standard_user\",\"permissions\":null,\"need_to_sign\":false,\"jwt\":{\"refresh\":\"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoicmVmcmVzaCIsImV4cCI6MTY0ODcyMTc2NCwianRpIjoiMjQxNjE0MmIxNzAwNDI4ZmE4YzNlYzczZmJkZWZkZGEiLCJ1c2VyX2lkIjo2NTgzLCJ1c2VybmFtZSI6InNhbWFuZHJ1c0BqdXBpdGVyb25lLmNvbSIsIm9yZ19pZCI6ImE3MTllOWE5LTY3OWUtNDgyMi1iNjNmLWU4NTJhZDI1ZmFlZCIsIm9yZ19uYW1lIjoiQWxvbmVzdCJ9.3APJrHmw0MiPJg73aHyk975R80VrmgqPq2bt6z19-U4\",\"access\":\"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjQ4NjM2MjY0LCJqdGkiOiIyZGRkMzJjYmQ3YTU0NDdkOTZmODk2MmI3MjQ0MWRkOCIsInVzZXJfaWQiOjY1ODMsInVzZXJuYW1lIjoic2FtYW5kcnVzQGp1cGl0ZXJvbmUuY29tIiwib3JnX2lkIjoiYTcxOWU5YTktNjc5ZS00ODIyLWI2M2YtZTg1MmFkMjVmYWVkIiwib3JnX25hbWUiOiJBbG9uZXN0In0.hVgib4uos4YV_4WT92vDGs37791VQsXoeMwUPkOP9cE\"},\"license_status\":2,\"license_hash\":\"ef85c9d7fcfeecfe4186067a6721dbaf4f183788fcf65cbf5930b1b6c9ad856d2df175e11b99e69065a7fd307dc7214714008b37f2a476cf3ab6d71f4033e0f2\",\"org_license_status\":2,\"org_license_current_hash\":\"ef85c9d7fcfeecfe4186067a6721dbaf4f183788fcf65cbf5930b1b6c9ad856d2df175e11b99e69065a7fd307dc7214714008b37f2a476cf3ab6d71f4033e0f2\"}"
          },
          "cookies": [
            {
              "expires": "2023-03-29T10:16:04.000Z",
              "maxAge": 31449600,
              "name": "csrftoken",
              "path": "/",
              "sameSite": "Lax",
              "value": "[REDACTED]"
            },
            {
              "expires": "2022-03-31T10:16:04.000Z",
              "httpOnly": true,
              "maxAge": 86400,
              "name": "sessionid",
              "path": "/",
              "sameSite": "Lax",
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 30 Mar 2022 10:16:04 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "722"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "gunicorn"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Cookie, Origin"
            },
            {
              "name": "allow",
              "value": "GET, POST, DELETE, HEAD, OPTIONS"
            },
            {
              "name": "x-orca-request-id",
              "value": "3c96a93c-0d94-427f-bd45-f4a73d5214d2"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-62442de4-2888211e5244d7297be90233"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "referrer-policy",
              "value": "same-origin"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            }
          ],
          "headersSize": 752,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-03-30T10:16:03.662Z",
        "time": 540,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 540
        }
      },
      {
        "_id": "2952034ba70eea17e58196a3bcdeb6f9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.orcasecurity.io"
            }
          ],
          "headersSize": 633,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.orcasecurity.io/api/organization/users"
        },
        "response": {
          "bodySize": 1723,
          "content": {
            "mimeType": "application/json",
            "size": 1723,
            "text": "{\"status\":\"success\",\"data\":{\"name\":\"Alonest\",\"users\":[{\"user_id\":\"d8c8ba01-2f7d-49c5-a000-3e4d55a36435\",\"email\":\"sumologic-internal-user-e852ad25faed@orca.security\",\"first\":\"sumologic\",\"last\":\"sumologic\"},{\"user_id\":\"32963ec9-7870-4b93-a210-6cbf152ec749\",\"email\":\"ssotest@Alonest.com\",\"first\":\"sso\",\"last\":\"test\"},{\"user_id\":\"06010e2d-5aa4-40ab-b569-30014db287bc\",\"email\":\"slack-internal-user-e852ad25faed@orca.security\",\"first\":\"slack\",\"last\":\"slack\"},{\"user_id\":\"95ae962e-4268-41f2-b9c6-33a387e957e4\",\"email\":\"demisto-internal-user-e852ad25faed@orca.security\",\"first\":\"demisto\",\"last\":\"demisto\"},{\"user_id\":\"2782bfd1-3d79-44a5-8f22-3169091debae\",\"email\":\"jira-internal-user-e852ad25faed@orca.security\",\"first\":\"jira\",\"last\":\"jira\"},{\"user_id\":\"a83a62f0-0a07-4ffc-b8e8-bb75fa52cf58\",\"email\":\"s3_bucket-internal-user-e852ad25faed@orca.security\",\"first\":\"s3_bucket\",\"last\":\"s3_bucket\"},{\"user_id\":\"c0b79156-b7e7-4d20-9c71-6e5bb58259e9\",\"email\":\"datadog-internal-user-e852ad25faed@orca.security\",\"first\":\"datadog\",\"last\":\"datadog\"},{\"user_id\":\"fb0b6183-f16e-42d8-907b-d117e31a1b55\",\"email\":\"alon+sso@orca.security\",\"first\":\"Alon\",\"last\":\"SSO\"},{\"user_id\":\"1244d386-5b94-4709-879a-5a7d738aea13\",\"email\":\"yuri+test@orca.security\",\"first\":\"Yuri\",\"last\":\"P\"},{\"user_id\":\"a81c2908-9c64-43eb-9497-e8712b9a9a72\",\"email\":\"pagerduty-internal-user-e852ad25faed@orca.security\",\"first\":\"pagerduty\",\"last\":\"pagerduty\"},{\"user_id\":\"c9ce76ce-1c08-45ec-9df3-c1be58e75bea\",\"email\":\"qradar-internal-user-e852ad25faed@orca.security\",\"first\":\"qradar\",\"last\":\"qradar\"},{\"user_id\":\"fc8af368-70cf-40ab-bb5d-da9ee3789397\",\"email\":\"demisto-pull-internal-user-e852ad25faed@orca.security\",\"first\":\"demisto-pull\",\"last\":\"demisto-pull\"},{\"user_id\":\"acb9ba3c-96da-494e-81eb-51123d90f953\",\"email\":\"orcaroles@ssotest.com\",\"first\":\"orca\",\"last\":\"roles\"},{\"user_id\":\"a7ac09bd-cf6f-496e-b13f-68e6b7d9c430\",\"email\":\"dimay1982@gmail.com\",\"first\":\"Dima\",\"last\":\"Yagudaev\"},{\"user_id\":\"4cceebbd-23bf-4043-bba8-9c2f2c43cdb0\",\"email\":\"alon+onelogin@orca.security\",\"first\":\"Same\",\"last\":\"Same\"},{\"user_id\":\"03833bae-8433-4b98-8ea9-7afdaeacc0b5\",\"email\":\"axonius-internal-user-e852ad25faed@orca.security\",\"first\":\"axonius\",\"last\":\"axonius\"},{\"user_id\":\"6b09306e-5169-413e-8635-08fdcd163850\",\"email\":\"opsgenie-internal-user-e852ad25faed@orca.security\",\"first\":\"opsgenie\",\"last\":\"opsgenie\"},{\"user_id\":\"fa62d286-0638-423f-b448-448a7baf029f\",\"email\":\"alon+cont@orca.security\",\"first\":\"Alon\",\"last\":\"Lavian\"},{\"user_id\":\"a25d69f2-4745-451b-8f30-8907b39b5726\",\"email\":\"yoav@orca.security\",\"first\":\"givenName\",\"last\":\"familyName\"},{\"user_id\":\"888d5d79-59db-42e0-a726-6287556708f1\",\"email\":\"webhook-internal-user-e852ad25faed@orca.security\",\"first\":\"webhook\",\"last\":\"webhook\"},{\"user_id\":\"a0d8d9a0-42ed-4281-8cd5-849beb7c5b19\",\"email\":\"yuri+slack_demo@orca.security\",\"first\":\"Slack\",\"last\":\"Demo\"},{\"user_id\":\"147c11c0-62bc-4558-8774-f785e542b5c6\",\"email\":\"azure_sentinel-internal-user-e852ad25faed@orca.security\",\"first\":\"azure_sentinel\",\"last\":\"azure_sentinel\"},{\"user_id\":\"82d4056a-0455-4f47-94f9-d039e84def2f\",\"email\":\"vitaly+neworg@orca.security\",\"first\":\"Vitali\",\"last\":\"Falileev\"},{\"user_id\":\"b2bbf98f-cc29-4661-adc4-feae2f6133e0\",\"email\":\"vitali.falileev@gmail.com\",\"first\":\"Vitali\",\"last\":\"Falileev\"},{\"user_id\":\"f1872eb3-cca6-45b1-aae5-acb66d994974\",\"email\":\"vitaly+alonest@orca.security\",\"first\":\"Vitaly\",\"last\":\"Alonest\"},{\"user_id\":\"90ef33e6-b719-47b8-85da-3e8bfe702695\",\"email\":\"amiravrm@gmail.com\",\"first\":\"Amir\",\"last\":\"Avram\"},{\"user_id\":\"e417f5e0-82d5-4fac-82c9-1b9739e863c2\",\"email\":\"gal+risk@orca.security\",\"first\":\"Gal\",\"last\":\"Lotem\"},{\"user_id\":\"d4403fb0-c42a-4833-8c40-115bcb67c2aa\",\"email\":\"zalan.daniel@gmail.com\",\"first\":\"daniel\",\"last\":\"zalan\"},{\"user_id\":\"3cc53616-d880-4a68-8bf1-24d2ad37591a\",\"email\":\"sam.andrus@jupiterone.com\",\"first\":\"Sam\",\"last\":\"Andrus\"},{\"user_id\":\"0020c001-a8fd-4d81-bf35-57aefc20b9b9\",\"email\":\"mitiak@gmail.com\",\"first\":\"Dima\",\"last\":\"Kovalchuk\"},{\"user_id\":\"8b9d2a90-5178-4986-b21e-7e8367720ab6\",\"email\":\"david@vulcan.io\",\"first\":\"David\",\"last\":\"Gruberger\"},{\"user_id\":\"19b4d57f-6973-4b91-9673-006d4dda2ece\",\"email\":\"splunk-internal-user-a719e9a9-679@orca.security\",\"first\":\"splunk\",\"last\":\"splunk\"},{\"user_id\":\"52d662c0-937a-4d79-9ada-6f7c9d267d2d\",\"email\":\"alon+test2@orca.security\",\"first\":\"Alon\",\"last\":\"Lavian\"},{\"user_id\":\"ab09f142-f9bc-4feb-85b5-e0a9b60b7d8a\",\"email\":\"guy@anecdotes.ai\",\"first\":\"Guy\",\"last\":\"Boyarski\"}],\"pending_invites\":[{\"email\":\"viktor+alonest@orca.security\",\"role\":\"Administrator\",\"all_cloud_account\":true,\"cloud_accounts\":[]},{\"email\":\"jayson.jensen@jupiterone.com\",\"role\":\"Viewer Without Export\",\"all_cloud_account\":false,\"cloud_accounts\":[\"AWS - 570398916848\"]},{\"email\":\"sela+alonest@orca.security\",\"role\":\"Administrator\",\"all_cloud_account\":true,\"cloud_accounts\":[]}]}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 30 Mar 2022 10:16:04 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "1723"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "gunicorn"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Cookie, Origin"
            },
            {
              "name": "allow",
              "value": "GET, POST, DELETE, HEAD, OPTIONS"
            },
            {
              "name": "x-orca-request-id",
              "value": "f109bb76-73e8-4ee9-b8c5-59d753a02642"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-62442de4-30984d01031a9cd63a8d16b8"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "referrer-policy",
              "value": "same-origin"
            }
          ],
          "headersSize": 444,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-03-30T10:16:04.210Z",
        "time": 451,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 451
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
