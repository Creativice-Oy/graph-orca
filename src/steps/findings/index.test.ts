import { executeStepWithDependencies } from '@jupiterone/integration-sdk-testing';
import { buildStepTestConfigForStep } from '../../../test/config';
import { Recording, setupProjectRecording } from '../../../test/recording';
import {
  Entities,
  MappedRelationships,
  Relationships,
  Steps,
} from '../constants';

// See test/README.md for details
let recording: Recording;
afterEach(async () => {
  await recording.stop();
});

jest.setTimeout(300000);

test('fetch-findings', async () => {
  recording = setupProjectRecording({
    directory: __dirname,
    name: 'fetch-findings',
  });

  const stepConfig = buildStepTestConfigForStep(Steps.FINDINGS);
  const { collectedEntities, collectedRelationships, encounteredTypes } =
    await executeStepWithDependencies(stepConfig);

  expect(encounteredTypes).toMatchSnapshot();
  expect(
    collectedEntities.some((e) => e._type === Entities.FINDING._type),
  ).toBeTruthy();
  expect(
    collectedRelationships.some(
      (e) => e._type === Relationships.ACCOUNT_HAS_FINDING._type,
    ),
  ).toBeTruthy();
  expect(
    collectedRelationships.some(
      (e) => e._type === Relationships.ASSET_HAS_FINDING._type,
    ),
  ).toBeTruthy();
  expect(
    collectedRelationships.some(
      (e) => e._type === MappedRelationships.FINDING_IS_CVE._type,
    ),
  ).toBeTruthy();
});
