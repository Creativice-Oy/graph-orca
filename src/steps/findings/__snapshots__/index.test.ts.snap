// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch-findings 1`] = `
Array [
  "orca_account",
  "orca_asset",
  "orca_account_has_asset",
  "orca_finding",
  "orca_account_has_finding",
  "orca_finding_is_cve",
  "orca_asset_has_finding",
]
`;
